File: SDKs/github.com/ghes-3.1/1.1.4/go/pkg/models/operations/gitgetblob.go
package operations

import (
"openapi/pkg/models/shared")

type GitGetBlobPathParams struct {
    FileSha string `pathParam:"style=simple,explode=false,name=file_sha"`
    Owner string `pathParam:"style=simple,explode=false,name=owner"`
    Repo string `pathParam:"style=simple,explode=false,name=repo"`
    
}

type GitGetBlobRequest struct {
    PathParams GitGetBlobPathParams 
    
}

type GitGetBlobResponse struct {
    ContentType string 
    StatusCode int64 
    BasicError *shared.BasicError 
    Blob *shared.Blob 
    ValidationError *shared.ValidationError 
    
}

File: SDKs/github.com/ghes-2.21/1.1.4/python/src/sdk/models/operations/git_get_blob.py
from dataclasses import dataclass, field
from typing import Optional
from sdk.models import shared


@dataclass
class GitGetBlobPathParams:
    file_sha: str = field(metadata={'path_param': { 'field_name': 'file_sha', 'style': 'simple', 'explode': False }})
    owner: str = field(metadata={'path_param': { 'field_name': 'owner', 'style': 'simple', 'explode': False }})
    repo: str = field(metadata={'path_param': { 'field_name': 'repo', 'style': 'simple', 'explode': False }})
    

@dataclass
class GitGetBlobRequest:
    path_params: GitGetBlobPathParams = field()
    

@dataclass
class GitGetBlobResponse:
    content_type: str = field()
    status_code: int = field()
    basic_error: Optional[shared.BasicError] = field(default=None)
    blob: Optional[shared.Blob] = field(default=None)
    validation_error: Optional[shared.ValidationError] = field(default=None)
    
